dlp/snippets/inspect/inspect_bigquery_with_sampling.go

```diff
@@ -0,0 +1,185 @@
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"cloud.google.com/go/dlp/apiv2/dlppb"
+	"cloud.google.com/go/pubsub"
+)
+
+// inspectBigQueryTableWithSampling inspect bigQueries for sensitive data with sampling
+func inspectBigQueryTableWithSampling(w io.Writer, projectID, topicID, subscriptionID string) error {
+	// projectId := "your-project-id"
+	// topicID := "your-pubsub-topic-id" or provide a topicID name to create one
+	// subscriptionID := "your-pubsub-topic-id" or provide a topicID name to create one
+
+	ctx := context.Background()
+
+	// Initialize a client once and reuse it to send multiple requests. Clients
+	// are safe to use across goroutines. When the client is no longer needed,
+	// call the Close method to cleanup its resources.
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Closing the client safely cleans up background resources.
+	defer client.Close()
+
+	// Specify the BigQuery table to be inspected.
+	tableReference := &dlppb.BigQueryTable{
+		ProjectId: "bigquery-public-data",
+		DatasetId: "usa_names",
+		TableId:   "usa_1910_current",
+	}
+
+	bigQueryOptions := &dlppb.BigQueryOptions{
+		TableReference: tableReference,
+		RowsLimit:      int64(10000),
+		SampleMethod:   dlppb.BigQueryOptions_RANDOM_START,
+		IdentifyingFields: []*dlppb.FieldId{
+			{Name: "name"},
+		},
+	}
+
+	// provide storage config with bigqueryOptions
+	storageConfig := &dlppb.StorageConfig{
+		Type: &dlppb.StorageConfig_BigQueryOptions{
+			BigQueryOptions: bigQueryOptions,
+		},
+	}
+
+	// Specify the type of info the inspection will look for.
+	// See https://cloud.google.com/dlp/docs/infotypes-reference for complete list of info types
+	infoTypes := []*dlppb.InfoType{
+		{Name: "PERSON_NAME"},
+	}
+
+	// Specify how the content should be inspected.
+	inspectConfig := &dlppb.InspectConfig{
+		InfoTypes:    infoTypes,
+		IncludeQuote: true,
+	}
+
+	// Create a PubSub Client used to listen for when the inspect job finishes.
+	pubsubClient, err := pubsub.NewClient(ctx, projectID)
+	if err != nil {
+		return err
+	}
+	defer pubsubClient.Close()
+
+	// Create a PubSub subscription we can use to listen for messages.
+	// Create the Topic if it doesn't exist.
+	t := pubsubClient.Topic(topicID)
+	if exists, err := t.Exists(ctx); err != nil {
+		fmt.Fprintf(w, "t.Exists: %v", err)
+		return err
+	} else if !exists {
+		if t, err = pubsubClient.CreateTopic(ctx, topicID); err != nil {
+			return err
+		}
+	}
+
+	// Create the Subscription if it doesn't exist.
+	s := pubsubClient.Subscription(subscriptionID)
+	if exists, err := s.Exists(ctx); err != nil {
+		return err
+	} else if !exists {
+		if s, err = pubsubClient.CreateSubscription(ctx, subscriptionID, pubsub.SubscriptionConfig{Topic: t}); err != nil {
+			return err
+		}
+	}
+
+	// topic is the PubSub topic string where messages should be sent.
+	topic := "projects/" + projectID + "/topics/" + topicID
```

Comment:
It's more idiomatic to use Sprintf:
`topic := fmt.Sprintf("projects/%s/topics/%s", projectID, topicID)`

Reply:
Updated it.